Options/ConfigProvider.php

```diff
@@ -23,26 +23,37 @@ class ConfigProvider implements ConfigProviderInterface
      */
     private $defaults;
 
-    public function __construct($defaults = [], $paths = [])
+    private $presets;
+
+    public function __construct($defaults = [], $paths = [], $presets = [])
     {
         $this->defaults = $defaults;
         $this->paths = $paths;
+        $this->presets = $presets;
     }
     /**
      * getPathConfig parses the request uri and attempt to match it against a path config, where the path config is used as a regex to match against the uri.
      * If no config is found, the default config is returned.
      * If a config is found, then the default and the path config is merged and returned in $options.
      * Path config is order dependant. IF '^/api' comes before '^/' in the config, then '^/api' will be applied and '^/' disregarded provided the first matches.
+     * Presets overwrite ALL config for a route, such that any config made in a route is ignored in favour of the defined configuration.
```

Comment:
I thought we were going the other way round, preset is like base template but you can provide specific route config to overrride it.

Reply:
Woops, I have to fix that comment because you're absolutely right and thats how the functionality works too!